pkg/config/state/user.go

```diff
@@ -46,7 +46,7 @@ func (u UserInfo) Validate() error {
 		),
 		validation.Field(&u.Username,
 			validation.Required,
-			validation.Match(regexp.MustCompile("^[a-z]{3}[0-9]{6}$")).Error("username must be in the form: yyyXXXXXX (y = letter, x = digit)"),
+			validation.Match(regexp.MustCompile("^[a-z]{3}[0-9]{4,7}$")).Error("username must be in the form: yyyXXXXXX (y = letter, x = digit)"),
```

Comment:
This validation is a duplicate of the one in aws.go, and I'm assuming they are supposed to always be the same.

Have you considered extracting them one place? `pkg/config/constant/api.go` would be a natural place. The benefit would be to avoid bugs because we forget to update one, but not the other.

Reply:
Did consider it, but didn't find the find a good pattern for it in the other pieces of the code. 

Let me fix that. It would be something like `api.ValidationOKUsername`

Reply:
@yngvark Updated with new constants and added tests for the MFAToken length
Some duplicate testSetup code, but they do test two different structs :)

Reply:
Looks good! Actually in a lot of cases I'd take duplication instead of the abstraction mess made by the Don't Repeat Yourself principle, so the testSetup code is just fine :+1: